tests/unit/libbpf_test.cpp

```diff
@@ -2978,6 +2978,17 @@ TEST_CASE("BPF_PROG_BIND_MAP etc.", "[libbpf]")
     attr.prog_bind_map.flags = 0;
     REQUIRE(bpf(BPF_PROG_BIND_MAP, &attr, sizeof(attr)) == 0);
 
+    // Verify we can create a nested map.
+    attr.map_create = {};
+    attr.map_create.map_type = BPF_MAP_TYPE_ARRAY_OF_MAPS;
+    attr.map_create.key_size = 4;
+    attr.map_create.value_size = 4;
+    attr.map_create.max_entries = 1;
+    attr.map_create.inner_map_fd = map_fd;
+    int nested_map_fd = bpf(BPF_MAP_CREATE, &attr, sizeof(attr.map_create));
```

Comment:
Can you add cases for update, lookup, and delete elements from the inner map?

Reply:
I added a single case which does a BPF_MAP_UPDATE_ELEM and then LOOKUP_ELEM in the outer map. This ensures that:

- The returned fd is valid
- Passing inner_map_fd is working
- Updating the nested map works
- Looking up a map returns the correct ID

I'm not sure what doing a delete would test, so I skipped that part.